src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = "::";
+    private static final String COMA = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final int ACTION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] arrStr = readFromFile(fromFileName);
+        writeToFile(toFileName, makeCounts(arrStr));
+    }
+
+    private String makeCounts(String[] arrStr) {
+        String[] tmpArrStr;
+        int buy = 0;
+        int supply = 0;
+        for (String s : arrStr) {
+            tmpArrStr = s.split(",");
+            switch (tmpArrStr[ACTION_INDEX]) {
+                case BUY:
+                    buy += Integer.parseInt(tmpArrStr[AMOUNT_INDEX]);
+                    break;
+                case SUPPLY:
+                    supply += Integer.parseInt(tmpArrStr[AMOUNT_INDEX]);
+                    break;
+                default:
+                    System.out.println("ОСЬ ТОБІ ДУЛЯ ..!..");
+            }
+        }
+        return SUPPLY + COMA + supply + System.lineSeparator()
+                + BUY + COMA + buy + System.lineSeparator()
+                + RESULT + COMA + (supply - buy);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        File fin = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder(4096);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fin))) {
```

Comment:
It is also better to divide this method into two. One will only write to the file and the other will generate the report